fibermiddleware_test.go

```diff
@@ -0,0 +1,394 @@
+		const statusCode = 200
+		const requestID = "my-req-id"
+
+		logger, _ := test.NewNullLogger()
+		hook := testMockFiberMiddlewareInvocation(func(c *fiber.Ctx) error {
+			c.Status(statusCode)
+			return nil
+		}, requestID, logger, "")
+
+		entries := hook.AllEntries()
+		assert.Equal(t, len(entries), 1, "Unexpected entries length.")
+
+		i := 0
+		outcomingRequest := entries[i]
+		logAssertions(t, outcomingRequest, ExpectedLogFields{
+			Level:     logrus.InfoLevel,
+			Message:   "request completed",
+			RequestID: requestID,
+		})
+		outcomingRequestAssertions(t, outcomingRequest, ExpectedOutcomingLogFields{
+			Method:        http.MethodGet,
+			Path:          path,
+			Hostname:      hostname,
+			ForwardedHost: clientHost,
+			Original:      userAgent,
+			IP:            ip,
+			StatusCode:    statusCode,
+			Bytes:         bodyBytes,
+		})
+
+		hook.Reset()
+	})
+
+	t.Run("test getHostname with request path without port", func(t *testing.T) {
+		const statusCode = 200
+		const requestID = "my-req-id"
+		var requestPathWithoutPort = fmt.Sprintf("http://%s/my-req", hostname)
+
+		logger, _ := test.NewNullLogger()
+		hook := testMockFiberMiddlewareInvocation(func(c *fiber.Ctx) error {
+			c.Status(statusCode)
+			return nil
+		}, requestID, logger, requestPathWithoutPort)
+
+		entries := hook.AllEntries()
+		assert.Equal(t, len(entries), 1, "Unexpected entries length.")
+
+		i := 0
+		outcomingRequest := entries[i]
+		logAssertions(t, outcomingRequest, ExpectedLogFields{
+			Level:     logrus.InfoLevel,
+			Message:   "request completed",
+			RequestID: requestID,
+		})
+		outcomingRequestAssertions(t, outcomingRequest, ExpectedOutcomingLogFields{
+			Method:        http.MethodGet,
+			Path:          path,
+			Hostname:      hostname,
+			ForwardedHost: clientHost,
+			Original:      userAgent,
+			IP:            ip,
+			StatusCode:    statusCode,
+			Bytes:         bodyBytes,
+		})
+
+		hook.Reset()
+	})
+
+	t.Run("test getHostname with request path with query", func(t *testing.T) {
+		const statusCode = 200
+		const requestID = "my-req-id"
+		const pathWithQuery = "/my-req?foo=bar&some=other"
+
+		logger, _ := InitHelper(InitOptions{
+			DisableHTMLEscape: true,
+		})
+		hook := testMockFiberMiddlewareInvocation(func(c *fiber.Ctx) error {
+			c.Status(statusCode)
+			return nil
+		}, requestID, logger, pathWithQuery)
+
+		entries := hook.AllEntries()
+		assert.Equal(t, len(entries), 1, "Unexpected entries length.")
+		byteEntry, err := entries[0].Bytes()
+		assert.NilError(t, err)
+		assert.Check(t, strings.Contains(string(byteEntry), `"url":{"path":"/my-req?foo=bar&some=other"}`))
+
+		hook.Reset()
+	})
+
+	t.Run("middleware correctly passing configured logger with request id from request header", func(t *testing.T) {
+		const statusCode = 200
+		const requestID = "my-req-id"
+
+		hook := testMockFiberMiddlewareInvocation(func(c *fiber.Ctx) error {
+			c.Status(statusCode)
```

Comment:
In this handler, with this assertion, we should not use the logger?

Reply:
This test has a wrong title, renamed to `do not log skipped paths`